### PR TITLE
HTTP/2 Spec

### PR DESCRIPTION
**Pull request by jomo**
Every HTTP/2 communication starts with `PRISM`. This actually made it into [RFC 7540, Section 3.5](https://tools.ietf.org/html/rfc7540#section-3.5).
That commit was made in June 2013 after the revelation of the PRISM surveillance program in Snowden's documents.

Read @jgrahamc's [blog post](http://blog.jgc.org/2015/11/the-secret-message-hidden-in-every.html) for more info.

### CODE REVIEW

**Comment by Bengt** [comment]
Woa.

**Comment by mehulkar** [comment]
hmm, I'm thinking that this hall of fame needs an Reddit style ELI5 section for each entry. I read the blog post and the linked HN discussion and as someone who knows nothing about http, I'm still a little unclear on what this actually is. 

Is adding the "PRISM" string into the http2 spec an indication of the http2 spec being compromised? Does that string mean anything? Or is it just a joke pointing to the Prism surveillance program?

**Comment by jomo** [comment]
It was added shortly after [PRISM](https://en.wikipedia.org/wiki/PRISM_%28surveillance_program%29) became well known to the public (via Snowden's leaked documents).
The IETF actually had quite [some discussion](http://www.ietf.org/mail-archive/web/httpbisa/current/msg14288.html) about how to design HTTP/2 after the leaks.

---

ELI5
It's much rather an acknowledgment and political statement by the IETF, protesting against global surveillance.
And everyone who wants to use HTTP/2 is forced to start the conversation with:

``` HTTP/2
PRI * HTTP/2.0

SM

```

or `PRISM` if you remove some code.

Maybe they also wanted to remind people that the NSA is watching you. You might think that's a bad place to put this reminder, but it's actually not: If you're reading some HTTP traffic, you're reminded that the NSA is reading it likewise.

**Comment by Bengt** [comment]
In any event the commit would be epic and worthy to be in the hall of fame. So technically, the is no immediate need for an ELI5 section in this project. I fear it would be challenging to find explanations of all the pull requests that anybody can understand. Some of them only make sense with a lot of knowledge about the domain or culture they come from. Note that with [explain XKCD](http://www.explainxkcd.com/wiki/index.php/Main_Page), the ELI5 lives in another project. Maybe that is also a possibility, here.

**Comment by jomo** [comment]
This project's wiki would be the perfect place for explanations. We could automatically generate a link to the wiki (which may or may not exist)

**Comment by mehulkar** [comment]
I get it now. Is this still part of the spec? If it is, this is definitely epic and worthy of hall of fame.

**Comment by jomo** [comment]
Yes it still is.
